File: SingleAPIExample/python/Cancel.py
import splitit

username, password, api_key = (
    "username",
    "password",
    "api_key",
)

api_client = splitit.ApiClient(splitit.Configuration(username, password, api_key, sandbox=True))


def cancel_installment_plan(ipn):
    resp = api_client.InstallmentPlanApi.installment_plan_cancel(
        splitit.CancelInstallmentPlanRequest(
            installment_plan_number=ipn,
            refund_under_cancelation=splitit.RefundUnderCancelation.ONLYIFAFULLREFUNDISPOSSIBLE,
        )
    )
    return resp

File: SingleAPIExample/java/Cancel.java
import com.splitit.*;
import com.splitit.sdk.model.*;
import java.math.BigDecimal;


public class Cancel {
    private String apiKey, username, password;
    public Cancel(String username, String password, String apiKey){
        this.username = username;
        this.password = password;
        this.apiKey = apiKey;
    }

    public void example(String ipn) throws ApiException{
        ApiClient apiClient = new ApiClient()
            .sandbox(true)
            .username(username)
            .password(password)
            .apiKey(apiKey);


        var result = apiClient.getInstallmentPlanApi().installmentPlanCancel(
            new CancelInstallmentPlanRequest()
            .installmentPlanNumber(ipn)
            .refundUnderCancelation(RefundUnderCancelation.ONLYIFAFULLREFUNDISPOSSIBLE)
        );
        System.out.println(result);
    }
}
